src/firebaseConnect.js

```diff
@@ -89,23 +97,40 @@ export const createFirebaseConnect = (storeKey = 'store') => (
   }
 
   FirebaseConnectWrapped.propTypes = {
-    dispatch: PropTypes.func.isRequired,
-    firebase: PropTypes.object.isRequired
+    _dispatch: PropTypes.func.isRequired,
+    _firebaseRef: PropTypes.object.isRequired
   }
 
   const HoistedComp = hoistStatics(FirebaseConnectWrapped, WrappedComponent)
 
-  const FirebaseConnect = props => (
-    <ReactReduxFirebaseContext.Consumer>
-      {firebase => (
-        <HoistedComp
-          firebase={firebase}
-          dispatch={firebase.dispatch}
-          {...props}
-        />
-      )}
-    </ReactReduxFirebaseContext.Consumer>
-  )
+  const FirebaseConnect = props => {
+    // Check that reserved props are not supplied to a FirebaseConnected
+    // component and if they are, throw an error so the developer can rectify
+    // this issue.
+    const clashes = Object.keys(props).filter(
+      k => !!RESERVED_PROPS.find(r => r === k)
+    )
+
+    if (clashes) {
```

Comment:
should be if (clashes.length), !![] will resolve as true

Reply:
Correct, I have fixed this already.